src/OpenLoco/src/GameCommands/Road/RemoveRoadStation.cpp

```diff
@@ -0,0 +1,163 @@
+            }
+            if (elRoad->roadId() != roadId)
+            {
+                continue;
+            }
+            // TODO: not seeing this check in disasm
+            if (elRoad->roadObjectId() != roadObjectId)
+            {
+                continue;
+            }
+            if (elRoad->sequenceIndex() != index)
+            {
+                continue;
+            }
+            return elRoad;
+        }
+        return nullptr;
+    }
+
+    // 0x0048D2AC
+    static currency32_t removeRoadStation(const RoadStationRemovalArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3(16, 16, 0));
+        bool updateStationTileRegistration = true;
+
+        // Get road at position
+        auto* initialElRoad = getElRoad(args.pos, args.rotation, args.type, args.roadId, args.index);
+        if (initialElRoad == nullptr)
+        {
+            return FAILURE;
+        }
+
+        // No station element on this road?
+        if (!initialElRoad->hasStationElement())
+        {
+            return FAILURE;
+        }
+
+        // Find the station element for an ownership check
+        {
+            // Station element must be the next element after the last consecutive road element
+            World::StationElement* stationEl = nullptr;
+            auto* elRoad = initialElRoad;
+            while (!elRoad->isLast())
+            {
+                auto* nextEl = elRoad->next();
+                elRoad = nextEl->as<World::RoadElement>();
+                if (elRoad != nullptr && elRoad->baseHeight() == args.pos.z)
+                {
+                    continue;
+                }
+
+                stationEl = nextEl->as<World::StationElement>();
+                break;
+            }
+
+            if (stationEl == nullptr)
+            {
+                return FAILURE;
+            }
+
+            // NB: vanilla would query owner from station struct, not the station element
+            if (!sub_431E6A(stationEl->owner(), reinterpret_cast<const World::TileElement*>(stationEl)))
+            {
+                return FAILURE;
+            }
+        }
+
+        const auto& roadPieces = World::TrackData::getRoadPiece(args.roadId);
+        const auto& argPiece = roadPieces[args.index];
+        const auto roadStart = args.pos - World::Pos3(Math::Vector::rotate(World::Pos2(argPiece.x, argPiece.y), args.rotation), argPiece.z);
+
+        StationId foundStationId = StationId::null;
+        currency32_t totalCost = 0;
+
+        for (auto& piece : roadPieces)
+        {
+            const auto roadLoc = roadStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ piece.x, piece.y }, args.rotation), piece.z };
+
+            auto* elRoad = getElRoad(roadLoc, args.rotation, args.type, args.roadId, piece.index);
+            if (elRoad == nullptr)
+                return FAILURE;
+
+            auto* nextEl = elRoad->next();
```

Comment:
this is still wrong its not getting the begin and end pointers of the road elements so the next element might not be the station.